pkgs/by-name/op/openthread-border-router/package.nix

```diff
@@ -0,0 +1,129 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  systemdLibs,
+  avahi,
+  protobuf,
+  jsoncpp,
+  boost,
+  libnetfilter_queue,
+  libnfnetlink,
+  nodejs,
+  buildNpmPackage,
+}:
+let
+  pname = "ot-br-posix";
+  version = "unstable-2024-10-18";
+
+  src = fetchFromGitHub {
+    owner = "openthread";
+    repo = "ot-br-posix";
+    rev = "76bfeef4e3e0242c235aa6bdf5480a0642883599";
+    hash = "sha256-VtekJD++PAse+yNmWzpO8lsyw0jnEsRFrzs/wK6VqqM=";
+    fetchSubmodules = true;
+  };
+
+  # hass-addons contains a few patches
+  hass-addons = fetchFromGitHub {
+    owner = "home-assistant";
+    repo = "addons";
+    rev = "583a62a69fa92ca8fdf9a2f298270a50bb3663a1";
+    hash = "sha256-u+lvU7mlJZqCCDIT4n7WnHyzAd0nZCh9Ddvqjs4SkHA=";
+  };
+
+  frontendModules = buildNpmPackage {
+    pname = "${pname}-frontend";
+    inherit version;
+    src = "${src}/src/web/web-service/frontend";
+    npmDepsHash = "sha256-7UVfPICyIbHEClpr3p7eDR46OUzS8mVf6P7phnDpVLk=";
+    dontNpmBuild = true;
+  };
+in
+stdenv.mkDerivation {
+
+  inherit pname version src;
+
+  # warning _FORTIFY_SOURCE requires compiling with optimization (-O)
+  env.NIX_CFLAGS_COMPILE = "-O";
+
+  patches = [
+    ./dont-install-systemd-units.patch
+    ./dont-use-boost-static-libs.patch
+    "${hass-addons}/openthread_border_router/0002-rest-support-deleting-the-dataset.patch"
+  ];
+
+  postPatch = ''
+    pushd third_party/openthread/repo
+      patch -p1 -i "${hass-addons}/openthread_border_router/0001-channel-monitor-disable-by-default.patch"
+    popd
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    nodejs
+  ];
+
+  # Adding npmConfigHook and manually passing fetchNpmDeps was resulting in ENOTCACHED errors
+  postConfigure = ''
+    ln -sf ${frontendModules}/lib/node_modules/otbr-web/node_modules ./src/web/web-service/frontend/
+  '';
+
+  buildInputs = [
+    avahi
+    systemdLibs
+    protobuf
+    jsoncpp
+    boost
+    libnetfilter_queue
+    libnfnetlink
+  ];
+
+  postInstall = ''
+    mkdir -p $out/bin
+    cp ${src}/script/otbr-firewall $out/bin/
+    chmod +x $out/bin/otbr-firewall
+
+    # Patch the firewall script so we can run it within the systemd start script
+    sed -i 's/THREAD_IF=.*//' $out/bin/otbr-firewall
+    sed -i 's/.*init-functions//' $out/bin/otbr-firewall
+    sed -i 's/.*vars\.sh//' $out/bin/otbr-firewall
```

Comment:
Can we use substituteInPlace or patches instead to notice if those break on changes? Also the patching should be done in postPatch.